### PR TITLE
[AGNTR-161] Add task to collect Agent release metrics

### PR DESCRIPTION
**Pull request by kacper-murzyn**
### What does this PR do?

This PR adds new task to the release group which generates basic set of the release metrics. It requires milestone, freeze date and release date to pass and produces set of numbers which are meant to be copied to [Agent release metrics](https://docs.google.com/spreadsheets/d/1r39CtyuvoznIDx1JhhLHQeAzmJB182n7ln8nToiWQ8s/edit#gid=1490566519).

There used to be scripts collecting those numbers but to simplify the usage I reviewed and moved the most important parts to the new invoke task.

This task is meant to be run once per release, so every 5 weeks.

### Motivation

Agent release process automation.

### Additional Notes

The most tricky part is the `get_prs_metrics` and why it was written this way. 
PyGithub which we use for all GH related activities is usually good enough but does not provide straight solution for this case when we need just the total number of PRs without getting details about them. If I use [get_pulls](https://github.com/DataDog/datadog-agent/blob/main/tasks/libs/ciproviders/github_api.py#L99) I hit the GH limit very often as there is almost no filter and we need to get information about 700-900 PRs, each one in the separate request. 
By making direct request to the GH endpoint and reading just a single field, whole procedure is quick and does not burden the server. I still used the PyGithub objects to reuse authentication approach.

### Describe how to test/QA your changes

Run the task and compare to the actual release metrics. [Agent release metrics](https://docs.google.com/spreadsheets/d/1r39CtyuvoznIDx1JhhLHQeAzmJB182n7ln8nToiWQ8s/edit#gid=1490566519.) can be used as a reference.

### CODE REVIEW

**Review by pducolin — APPROVED** [review]

File: tasks/release.py [hunk 1]
```diff
@@ -1762,3 +1763,38 @@ def get_active_release_branch(_ctx):
...
+    print("--------------------------")
+    print(lead_time)
+
+    # Step 2: Agent stability data
+    prs = get_prs_metrics(milestone, freeze_date)
+    print("Agent stability data")
+    print("--------------------")
+    print(f"{prs['total']}, {prs['before_freeze']}, {prs['on_freeze']}, {prs['after_freeze']}")
+
+    # Step 2: Code changes
```
Comment (KevinFairise2) [inline tasks/release.py:1793]: **🥜 nitpick:** ‏
```suggestion
    # Step 3: Code changes
```

**Review by KevinFairise2 — APPROVED** [review]

File: tasks/libs/ciproviders/github_api.py [hunk 2]
```diff
@@ -34,6 +34,13 @@ def repo(self):
 
         return self._repository
 
+    def raw_request(self, url):
```
Comment (chouetz) [inline tasks/libs/ciproviders/github_api.py:37]: I don't think we need this, I give some code examples below

File: tasks/release_metrics/metrics.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    github = GithubAPI(repository=GITHUB_REPO_NAME)
+
+    pr_counts = {"total": 0}
+
+    for key, url in QUERY_URLS.items():
+        _, prs = github.raw_request(url.format(milestone, freeze_date))
+        pr_counts[key] = prs["total_count"]
+        pr_counts["total"] += pr_counts[key]
+
+    return pr_counts
```
Comment (chouetz) [inline tasks/release_metrics/metrics.py:36]: Code is more verbose using the existing methods of the module but it's possible
```suggestion
def get_prs_metrics(milestone, freeze_date):
    github = GithubAPI(repository=GITHUB_REPO_NAME)
    repo = github.repo()
    pr_counts = {"total": 0, "before_freeze": 0, "on_freeze": 0, "after_freeze": 0}
    m = get_milestone(repo, milestone)
    issues = repo.get_issues(m, state='closed')
    for issue in issues:
        if issue.pull_request is None or issue.pull_request.raw_data['merged_at'] is None:
            continue
        # until 3.11 we need to strip the date string
        merged = datetime.fromisoformat(issue.pull_request.raw_data['merged_at'][:-1])
        if merged < freeze_date:
            pr_counts["before_freeze"] += 1
        elif merged = freeze_date:
            pr_counts["on_freeze"] += 1
        else:
            pr_counts["after_freeze"] += 1

    pr_counts["total"] = pr_counts["before_freeze"] + pr_counts["on_freeze"] + pr_counts["after_freeze"]

    return pr_counts

def get_milestone(repo, milestone):
    milestones = repo.get_milestones(state="all")
    for mile in milestones:
        if mile.title == milestone:
            return mile
    return None
```

**Review by chouetz — APPROVED** [review]
The module is maybe not the more convenient in this case, I let you select what/how you want to do

File: tasks/release.py (hunk 1 above)
Comment (kacper-murzyn) [reply tasks/release.py]: Corrected. Thanks for spotting that

File: tasks/libs/ciproviders/github_api.py (hunk 2 above)
Comment (kacper-murzyn) [reply tasks/libs/ciproviders/github_api.py]: Ok, I actually think it is better as it avoids that extra method added to the api class. It is still slower and maybe harder to follow but at least in one place. I had to tweak your suggestion a little but tested it and it works good. Thanks!

File: tasks/release.py [hunk 4]
```diff
@@ -1762,3 +1763,38 @@ def get_active_release_branch(_ctx):
...
+    code_stats = ctx.run(
+        f"git log --shortstat {milestone}-devel..{milestone} | grep \"files changed\" | awk '{{files+=$1; inserted+=$4; deleted+=$6}} END {{print files,\",\", inserted,\",\", deleted}}'",
```
[the remark this answers is not in the document]
Comment (kacper-murzyn) [reply tasks/release.py:1796]: The actual scope of the release is between `7.XX.0-devel` and `7.XX.0` tags. `devel` is created on the main branch after we create the release branch. The `7.XX.0` are always on different branches.

**Pull request merged**